src/java/org/apache/cassandra/index/sai/disk/v1/WideRowAwarePrimaryKeyMap.java

```diff
@@ -0,0 +1,178 @@
+
+        protected FileHandle partitionsFile;
+        protected FileHandle clusteringKeyBlockOffsetsFile;
+        protected FileHandle clustingingKeyBlocksFile;
+
+        public WideRowAwarePrimaryKeyMapFactory(IndexDescriptor indexDescriptor, SSTableReader sstable)
+        {
+            super(indexDescriptor, sstable);
+
+            try
+            {
+                this.clusteringComparator = indexDescriptor.clusteringComparator;
+                this.partitionsFile = indexDescriptor.createPerSSTableFileHandle(IndexComponent.PARTITION_SIZES);
+                this.clusteringKeyBlockOffsetsFile = indexDescriptor.createPerSSTableFileHandle(IndexComponent.CLUSTERING_KEY_BLOCK_OFFSETS);
+                this.clustingingKeyBlocksFile = indexDescriptor.createPerSSTableFileHandle(IndexComponent.CLUSTERING_KEY_BLOCKS);
+                NumericValuesMeta partitionsMeta = new NumericValuesMeta(metadataSource.get(indexDescriptor.componentName(IndexComponent.PARTITION_SIZES)));
+                this.partitionReaderFactory = new MonotonicBlockPackedReader(partitionsFile, partitionsMeta);
+                NumericValuesMeta clusteringKeyBlockOffsetsMeta = new NumericValuesMeta(metadataSource.get(indexDescriptor.componentName(IndexComponent.CLUSTERING_KEY_BLOCK_OFFSETS)));
+                this.clusteringKeyMeta = new SortedTermsMeta(metadataSource.get(indexDescriptor.componentName(IndexComponent.CLUSTERING_KEY_BLOCKS)));
+                this.clusteringKeyReader = new SortedTermsReader(clustingingKeyBlocksFile, clusteringKeyBlockOffsetsFile, clusteringKeyMeta, clusteringKeyBlockOffsetsMeta);
+            }
+            catch (Throwable t)
+            {
+                throw Throwables.unchecked(Throwables.close(t, partitionsFile, clustingingKeyBlocksFile, clusteringKeyBlockOffsetsFile));
+            }
+        }
+
+        @Override
+        @SuppressWarnings({ "resource", "RedundantSuppression" })
+        public PrimaryKeyMap newPerSSTablePrimaryKeyMap() throws IOException
+        {
+            LongArray rowIdToToken = new LongArray.DeferredLongArray(tokenReaderFactory::open);
+            LongArray partitionIdToToken = new LongArray.DeferredLongArray(partitionReaderFactory::open);
+
+            return new WideRowAwarePrimaryKeyMap(rowIdToToken,
+                                                 partitionIdToToken,
+                                                 partitionKeyReader.openCursor(),
+                                                 clusteringKeyReader.openCursor(),
+                                                 partitioner,
+                                                 primaryKeyFactory,
+                                                 clusteringComparator);
+        }
+
+        @Override
+        public void close()
+        {
+            super.close();
+            FileUtils.closeQuietly(Arrays.asList(partitionsFile, clustingingKeyBlocksFile, clusteringKeyBlockOffsetsFile));
```

Comment:
`partitionsFile` is already closed in the `super.close()`, right?

Reply:
Correct, I've fixed this.